### ISSUE TITLE
Partial serverside BASS support

### ISSUE THREAD

**Issue by wyozi** [Facepunch/garrysmod-requests#439]
To for example get duration of an mp3 or #431 if it gets implemented.

**Comment by wyozi** [comment]
If someone is in desperate need of these, I implemented both ID3 and mp3 duration parsers in pure lua

https://github.com/wyozi/gmod-medialib/blob/master/ext/id3parser.lua
https://github.com/wyozi/gmod-medialib/blob/master/ext/mp3duration.lua

**Comment by Grocel** [comment]
It would be also useful for getting the FFT spectrum server side.

And this is getting implemented then you wouldn't have to deal with that mp3 tags trouble again: https://github.com/Facepunch/garrysmod-requests/issues/172
So far I have tested it in the past, it support every sound format that can hold tags.

In my experience (http://facepunch.com/showthread.php?t=1159184) BASS works just fine on shared use if you don't try to load it twice on listen servers or single player. The single instance can be accessed from both sides easily.

**Comment by thegrb93** [comment]
Would also be nice if it were possible to retrieve whether a .wav file should loop by default or not since some game sounds are loops.

**Comment by Grocel** [comment]
Yeah, unfortunately BASS (2.4.10) doesn't like those files and gives out an error -1. Also the looping information isn't saved as a tag. It is caused by named cues at the start and end of the file you add in with a sound editor.

**Comment by Vurv78** [comment]
6 years later and just want to let everyone know this is still wanted, since right now I have to make a huge convoluted net system for fft and audio stream length, current playback time for my addon..

**Comment by KrzOkun** [comment]
Bump
